pkg/splunk/enterprise/clustermanager.go

```diff
@@ -434,3 +451,160 @@ func VerifyCMisMultisite(ctx context.Context, cr *enterpriseApi.ClusterManager,
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// getClusterManagerCurrentImage gets the image of the pods of the clusterManager before any upgrade takes place,
+// returns the image, and error if something goes wring
+func getClusterManagerCurrentImage(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.ClusterManager) (string, error) {
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("getClusterManagerCurrentImage").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(SplunkClusterManager, cr.GetName()),
+	}
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil {
+		eventPublisher.Warning(ctx, "getClusterManagerCurrentImage", fmt.Sprintf("Could not get Stateful Set. Reason %v", err))
+		scopedLog.Error(err, "StatefulSet types not found in namespace", "namsespace", cr.GetNamespace())
+		return "", err
+	}
+	labelSelector, err := metav1.LabelSelectorAsSelector(statefulSet.Spec.Selector)
+	if err != nil {
+		eventPublisher.Warning(ctx, "getClusterManagerCurrentImage", fmt.Sprintf("Could not get labels. Reason %v", err))
+		scopedLog.Error(err, "Unable to get labels")
+		return "", err
+	}
+
+	// get a list of all pods in the namespace with matching labels as the statefulset
+	statefulsetPods := &corev1.PodList{}
+	opts := []rclient.ListOption{
+		rclient.InNamespace(cr.GetNamespace()),
+		rclient.MatchingLabelsSelector{Selector: labelSelector},
+	}
+
+	err = c.List(ctx, statefulsetPods, opts...)
+	if err != nil {
+		eventPublisher.Warning(ctx, "getClusterManagerCurrentImage", fmt.Sprintf("Could not get Pod list. Reason %v", err))
+		scopedLog.Error(err, "Pods types not found in namespace", "namsespace", cr.GetNamespace())
+		return "", err
+	}
+
+	// find the container with the phrase 'splunk' in it
+	for _, v := range statefulsetPods.Items {
+		for _, container := range v.Status.ContainerStatuses {
+			if strings.Contains(container.Name, "splunk") {
+				image := container.Image
+				return image, nil
+			}
+
+		}
+	}
+
+	return "", nil
+}
+
+// changeClusterManagerAnnotations updates the checkUpdateImage field of the CLuster Manager Annotations to trigger the reconcile loop
+// on update, and returns error if something is wrong
+func changeClusterManagerAnnotations(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.LicenseManager) error {
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("changeClusterManagerAnnotations").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	eventPublisher, _ := newK8EventPublisher(c, cr)
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      cr.Spec.ClusterManagerRef.Name,
+	}
+	clusterManagerInstance := &enterpriseApi.ClusterManager{}
```

Comment:
Do we always configure a `ClusterManagerRef` on LM? If not, we can list all CM's in the namespace and find the one with the correct LM Ref

Reply:
not necessarily, we may/may not configure CM with LM

Reply:
In that case, we should list all CM's and find the one folding LMRef here?